js/views/message_view.js

```diff
@@ -498,18 +510,21 @@
       // If we can't verify a signal account locally, we'll go to the Signal Server.
       if (number && !hasLocalSignalAccount) {
         // eslint-disable-next-line more/no-then
-        window.textsecure.messaging
+        this.checkingProfile = window.textsecure.messaging
           .getProfile(number)
           .then(() => {
+            this.contactHasSignalAccount = true;
```

Comment:
I understand that mutation of an instance variable is likely necessary for memoization, but could we encapsulate it somewhat from the render code?